### DISCUSSION TITLE
Trying to vertically align a label inside a Horizontal container

### DISCUSSION THREAD

**Discussion by Banbury** [Textualize/textual#4809] [Q&A] [answered]
Why does the label not align correctly? It's always stuck in the left upper corner.
```python
class TextInput(Static):
    DEFAULT_CSS = """
        #label {
            align: left middle;
            height: 1;
        }
    """

    def compose(self) -> ComposeResult:
        with Horizontal():
            yield Label("Label", id="label")
            yield Input()
```

**Comment by TomJGooding** [comment, marked as answer]
`align` works on the children of a container, it isn't something you use on the child you want centered.

https://textual.textualize.io/FAQ/#how-do-i-center-a-widget-in-a-screen
https://textual.textualize.io/how-to/center-things

**Reply by Banbury** [reply]
Thank you.
